hydrography-approach/run-hydrography-pipeline.py

```diff
@@ -13,146 +15,204 @@
 from processing_scripts.filter_data import filter_osm_ways, process_filter_nbi_bridges
 from processing_scripts.tag_data import tag_nbi_and_osm_data
 
+# Initialize global logger
+logger = logging.getLogger(__name__)
 
-def load_config(state_name):
+
+def load_config(state_name: str) -> Dict:
     """
-    Load configuration
+    Load and render configuration from a YAML file.
     """
     env = Environment(
         loader=FileSystemLoader("."), autoescape=select_autoescape(["yaml"])
     )
 
     with open("config.yml", "r") as file:
         template = env.from_string(file.read())
-        rendered_yaml = template.render(state=f"{state_name}")
+        rendered_yaml = template.render(state=state_name)
         config = yaml.safe_load(rendered_yaml)
 
     return config
 
 
-def main():
-    # Mention state to process
-    state_name = "Kentucky"
-
-    # Load config file
-    print("\nLoading the config file.")
-    config = load_config(state_name)
+def create_directories(config: Dict) -> None:
+    """
+    Create directories for output data as specified in the configuration.
+    """
+    output_folders = config["output_data_folders"]
+    for folder in ["state_folder", "csv_files", "gpkg_files", "pbf_files"]:
+        os.makedirs(output_folders[folder], exist_ok=True)
+    logger.info("Directories created.")
 
-    # Make the required directories for storing outputs
-    print("\nMake the required directories.")
-    os.makedirs(
-        config["output_data_folders"]["state_folder"],
-        exist_ok=True,
-    )
-    os.makedirs(config["output_data_folders"]["csv_files"], exist_ok=True)
-    os.makedirs(config["output_data_folders"]["gpkg_files"], exist_ok=True)
-    os.makedirs(config["output_data_folders"]["pbf_files"], exist_ok=True)
 
-    # --------------------------------------------Filter OSM ways data--------------------------------------------
+def filter_osm_data(config: Dict) -> None:
+    """
+    Filter OSM ways data using the provided configuration.
+    """
     input_osm_pbf = config["input_data_folder"]["state_latest_osm"]
     output_osm_pbf = config["output_files"]["filtered_osm_pbf"]
     output_gpkg = config["output_files"]["filtered_highways"]
 
-    print("\nFiltering OSM ways data.")
-    filter_osm_ways.filter_ways(input_osm_pbf, output_osm_pbf, output_gpkg)
+    logger.info("Filtering OSM ways data.")
```

Comment:
@varun-andhra-mapup can you refer to our meeting again to see how to close requests from PR agent review

Reply:
I have made changes based on PR feedback and posted change comments as well.